seeker.go

```diff
@@ -0,0 +1,192 @@
+package roundtrip
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gravitational/trace"
+)
+
+const (
+	stateOK = iota
+	stateEOF
+	stateError
+	stateClosed
+)
+
+// Seeker is a file download object
+// that supports io.ReadSeeker interface
+type seeker struct {
+	client        *Client
+	endpoint      string
+	fileSize      int64
+	currentReader io.ReadCloser
+	currentOffset int64
+	lastError     error
+	state         int
+}
+
+func (s *seeker) updateState(err error) error {
+	switch s.state {
+	case stateError:
```

Comment:
it can be even as simple as:
```go
func (s *seeker) updateState(err error) error {
	switch s.state {
	case stateOK, stateEOF:
        if err == io.EOF {
            s.state = stateEOF
        } else if err != nil {
            s.state = stateError
        }
        s.lastError = err
	}
	return err
}
```